src/builder/QuarkBuilder.sol

```diff
@@ -1308,6 +1308,262 @@ contract QuarkBuilder {
+        }
+
+        // Auto-wrap
+        checkAndInsertWrapOrUnwrapAction(
+            actions,
+            quarkOperations,
+            chainAccountsList,
+            payment,
+            supplyIntent.assetSymbol,
+            supplyIntent.amount,
+            supplyIntent.chainId,
+            supplyIntent.sender,
+            supplyIntent.blockTimestamp,
+            useQuotecall
+        );
+
+        (IQuarkWallet.QuarkOperation memory supplyQuarkOperation, Actions.Action memory supplyAction) = Actions
+            .morphoVaultSupply(
+            Actions.MorphoVaultSupply({
+                chainAccountsList: chainAccountsList,
+                assetSymbol: supplyIntent.assetSymbol,
+                amount: supplyIntent.amount,
+                blockTimestamp: supplyIntent.blockTimestamp,
+                chainId: supplyIntent.chainId,
+                sender: supplyIntent.sender
+            }),
+            payment,
+            useQuotecall
+        );
+
+        List.addQuarkOperation(quarkOperations, supplyQuarkOperation);
+        List.addAction(actions, supplyAction);
+
+        // Convert actions and quark operations to array
+        Actions.Action[] memory actionsArray = List.toActionArray(actions);
+        IQuarkWallet.QuarkOperation[] memory quarkOperationsArray = List.toQuarkOperationArray(quarkOperations);
+
+        // Validate generated actions for affordability
+        if (payment.isToken) {
+            assertSufficientPaymentTokenBalances(
+                actionsArray, chainAccountsList, supplyIntent.chainId, supplyIntent.sender
+            );
+        }
+
+        // Merge operations that are from the same chain into one Multicall operation
+        (quarkOperationsArray, actionsArray) =
+            QuarkOperationHelper.mergeSameChainOperations(quarkOperationsArray, actionsArray);
+
+        // Wrap operations around Paycall/Quotecall if payment is with token
+        if (payment.isToken) {
+            quarkOperationsArray = QuarkOperationHelper.wrapOperationsWithTokenPayment(
+                quarkOperationsArray, actionsArray, payment, useQuotecall
+            );
+        }
+
+        return BuilderResult({
+            version: VERSION,
+            actions: actionsArray,
+            quarkOperations: quarkOperationsArray,
+            paymentCurrency: payment.currency,
+            eip712Data: EIP712Helper.eip712DataForQuarkOperations(quarkOperationsArray, actionsArray)
+        });
+    }
+
+    struct MorphoVaultWithdrawIntent {
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        uint256 chainId;
+        address withdrawer;
+    }
+
+    function morphoVaultWithdraw(
+        MorphoVaultWithdrawIntent memory withdrawIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory) {
+        // XXX confirm that you actually have the amount to withdraw
```

Comment:
Is this still a to-do?

Reply:
Yeah this is copied from the cometWithdraw builder, and since both are mostly the same, so I'm pretty sure if cometWithdraw still have this `XXX` morpho will have the same as well. 

From my understanding this is an optional feature that it's left undecided from the beginning from cometWithdraw. 
Which we make builder to verify the input and revert if client ever pass in a larger amount to withdraw while user position doesn't have that much of asset in chainAccountList. 

But also we don't really need to check, as client won't likely to send in a request to withdraw larger amount than user has. Plus client also provide the ChainAccountList,  so if something very wrong in client causing client thinks user has more to withdraw than user has, it's likely client may also mess up the ChainAccountList as well which builder won't be able to catch either.